codalab/apps/web/models.py

```diff
@@ -2315,11 +2324,17 @@ class OrganizerDataSet(models.Model):
     sub_data_files = models.ManyToManyField('OrganizerDataSet', null=True, blank=True, verbose_name="Bundle of data files")
     uploaded_by = models.ForeignKey(settings.AUTH_USER_MODEL)
     key = UUIDField(version=4)
+    is_public = models.BooleanField(default=False)
+    created_when = models.DateTimeField(null=True, blank=True)
```

Comment:
should we do `default=now` instead of checking in the save?

Reply:
I do not want to set default on old models so I'm a bit hesitant to do that. When we do this migration, I think it will set ALL old datasets `created_when` to `now()`

Reply:
if those old instances ever get updated, their `created_when` will get set to `now` also, but yea, probably fewer will be affected that way.

Reply:
yeah that is true... maybe I could do some clever stuff and get "last modified" from the file or something? =\ probably not worth it

Reply:
could we check for a pk in save? and only write this value on create?

Reply:
added check for PK